internal/core/worktree.go

```
@@ -28,6 +34,29 @@ func NewWorktreeManager(gitRepo git.Repository, configManager *config.Manager) *
 	}
 }
 
+// SetEventObserver registers a callback that fires for each hook phase event
+// (including the synthetic interrupted event on non-zero exit) as it is
+// parsed live. Pass nil to clear. The callback must not block the caller;
+// forward events to a buffered channel if downstream work is slow.
+func (wm *WorktreeManager) SetEventObserver(fn func(events.Event)) {
+	// atomic.Value requires concrete type consistency, so store a typed nil.
+	var v func(events.Event) = fn
```

golangci-lint (2.11.4)

[error] ST1023: should omit type func(events.Event) from declaration; it will be inferred from the right-hand side

(staticcheck)